src/diff_edit.rs

```diff
@@ -96,6 +135,122 @@ fn set_readonly_recursively(path: &Path) -> Result<(), std::io::Error> {
     }
 }
 
+pub fn run_mergetool(
+    _ui: &mut Ui,
+    _settings: &UserSettings,
+    tree: &Tree,
+    repo_path: &RepoPath,
+) -> Result<TreeId, DiffEditError> {
+    let conflict_id = match tree.path_value(repo_path) {
+        Some(TreeValue::Conflict(id)) => id,
+        Some(_) => return Err(DiffEditError::NotAConflictError(repo_path.clone())),
+        None => return Err(DiffEditError::PathNotFoundError(repo_path.clone())),
+    };
+    let conflict = tree.store().read_conflict(repo_path, &conflict_id)?;
+    let mut materialized_conflict: Vec<u8> = vec![];
+    materialize_conflict(
+        tree.store(),
+        repo_path,
+        &conflict,
+        &mut materialized_conflict,
+    )
+    .unwrap_or_else(|_| materialized_conflict.clear());
+    // For normal files, this is the default user-facing represenation of conflicts
+    // as a text file with conflict markers.
+    let materialized_conflict = materialized_conflict;
+
+    let content = match extract_file_conflict_data(tree.store(), repo_path, &conflict) {
+        Some(c) => c,
+        _ => {
+            // In this case, `materialized_conflict` is a short summary of what kind of
+            // files are involved in the conflict.
```

Comment:
Instead of depending on this, what do you think about calling `describe_conflict()` here and calling your new `materialize_merge_result()` in the successful case? That way we don't depend on what `materialized_conflict` does.

Reply:
I can do that. I think the reason I didn't do this initially is that it means I need to make `describe_conflict` public. That's probably a good idea regardless.

Reply:
I haven't thought much about what should or should not be public yet, so if `describe_conflict` is currently not public, it's probably just because it didn't need to be yet :)